tests/jballerina-unit-test/src/test/resources/test-src/expressions/binaryoperations/add-operation.bal

```diff
@@ -140,11 +140,53 @@ function testContextuallyExpectedTypeOfNumericLiteralInAdd() {
     float a2 = 5 + 3 + 10.0;
     decimal a3 = 5 + 15.0;
     decimal a4 = 5.0 + 10.0 + 10;
+    float? a5 = 10 + 5;
+    decimal? a6 = 5 + 3;
 
     assertEqual(a1, 15.0);
     assertEqual(a2, 18.0);
     assertEqual(a3, 20.0d);
     assertEqual(a4, 25.0d);
+    assertEqual(a5, 15.0);
+    assertEqual(a6, 8.0d);
+}
+
+type Ints 1|2;
+
+function testAddNullable() {
+    int? a1 = 5;
+    int? a2 = 6;
+    int? a3 = 10;
+    int? a4 = ();
+    int a5 = 12;
+    float? a6 = 5.5;
+    float? a7 = 10.0;
+    float? a8 = ();
+    float a9 = 5.0;
+
+    int? a10 = a1 + a2 + a3 + a5;
+    int? a11 = a5 + a3;
+    int? a12 = a4 + a1;
+    float? a13 = a6 + a7;
+    float? a14 = a6 + a9;
+    float? a15 = a6 + a8;
```

Comment:
Shall we add tests for decimal cases as well? for all other operations (add, multiple,.. etc)

Reply:
Please check the below case also. Seems not working

```
type Decimals -2.0d|-1.0d|0.0d|1.0d|2.0d|();

function testMultiplyNullable() {

    Decimals a = 1.0d;
    Decimals b = 2.0d;
    decimal? c = a / a;
    decimal? d = a / b;
}
```

Reply:
Will add different permutations of `decimal` test cases.